webwhatsapi/js/wapi.js

```diff
@@ -1147,3 +1148,27 @@ window.WAPI.getBufferedNewMessages = function(done) {
     return bufferedMessages;
 };
 /** End new messages observable functions **/
+
+window.WAPI.sendImage = function (imgBase64, chatid, filename, caption, done) {
+    var chat = WAPI.getChat(chatid);
+    if (chat !== undefined) {
+        var mediaBlob = window.WAPI.base64ImageToFile(imgBase64, filename);
+        Store.MediaCollection.processFiles([mediaBlob], chat, 1).then(() => {
+            var media = Store.MediaCollection.models[Store.MediaCollection.models.length - 1];
+            media.sendToChat(chat, {caption: caption});
+            done(true);
+        });
+    } else {
+        done(false);
+    }
+    return true;
+};
+
+window.WAPI.base64ImageToFile = function (b64Data, filename) {
+    var arr = b64Data.split(','), mime = arr[0].match(/:(.*?);/)[1],
+            bstr = atob(arr[1]), n = bstr.length, u8arr = new Uint8Array(n);
+    while (n--) {
+        u8arr[n] = bstr.charCodeAt(n);
+    }
+    return new File([u8arr], filename, {type: mime});
```

Comment:
have you tested with large files?

Reply:
no.. max size i tested is 331k... i will test now on large files

Reply:
@alfred82santa tested now with 16mb file... works fine!

Reply:
https://stackoverflow.com/questions/16245767/creating-a-blob-from-a-base64-string-in-javascript#answer-16245768